seqr/views/utils/dataset_utils.py

```diff
@@ -261,28 +263,41 @@ def match_and_update_search_samples(
 
 
 def _match_and_update_rna_samples(
-    projects, user, sample_ids, data_source, sample_id_to_individual_id_mapping, raise_unmatched_error_template,
+    projects, user, sample_project_tuples, data_source, sample_id_to_individual_id_mapping, raise_unmatched_error_template,
 ):
-    samples = Sample.objects.select_related('individual').filter(
+    sample_project_lookup = defaultdict(set)
+    for sample_id, project_name in sample_project_tuples:
+        sample_project_lookup[sample_id].add(project_name)
+    no_project_key = next(iter(list(sample_project_tuples)), ('', None))[1] is None
+    samples = Sample.objects.select_related('individual__family__project').filter(
         individual__family__project__in=projects,
         sample_type=Sample.SAMPLE_TYPE_RNA,
         dataset_type=Sample.DATASET_TYPE_VARIANT_CALLS,
-        sample_id__in=sample_ids,
-    )
+        sample_id__in=sample_project_lookup.keys(),
+    ).annotate(project_name=Value(None, output_field=TextField()) if no_project_key else F('individual__family__project__name'))
```

Comment:
After updating Outlier, it can be simplified to `annotate(project_name=F('individual__family__project__name'))`.